lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,533 @@
+      checksum_type: null,
+      file_size: null,
+      file_name: null,
+      key: null,
+      source: null,
+    }
+  );
+});
+
+test.serial('migrateGranulesAndFiles skips already migrated granule record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await Promise.all(testGranule.files.map((file) => s3Utils.s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: 'some-body',
+  })));
+  await migrateGranuleRecord(testGranule, knex);
+  await granulesModel.create(testGranule);
+
+  t.teardown(() => {
+    granulesModel.delete(testGranule);
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 0,
+      failed: 0,
+      skipped: 0,
+      success: 0,
+    },
+    granulesSummary: {
+      dynamoRecords: 1,
+      failed: 0,
+      skipped: 1,
+      success: 0,
+    },
+  });
+
+  const records = await knex(tableNames.granules);
+  t.is(records.length, 1);
+});
+
+test.serial('migrateGranulesAndFiles processes multiple granules and files', async (t) => {
+  const {
+    collectionPgModel,
+    executionPgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const alternateBucket = cryptoRandomString({ length: 10 });
+  await s3Utils.createBucket(alternateBucket);
+
+  const testExecution2 = fakeExecutionRecordFactory();
+  await executionPgModel.create(
+    knex,
+    testExecution2
+  );
+
+  const testCollection2 = fakeCollectionRecordFactory();
+  await collectionPgModel.create(
+    t.context.knex,
+    testCollection2
+  );
+
+  const testGranule1 = testGranule;
+  const testGranule2 = generateTestGranule(testCollection2, testExecution2.arn, alternateBucket);
+  const files = testGranule1.files.concat(testGranule2.files);
+
+  await Promise.all(files.flatMap((file) => s3Utils.s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: 'some-body',
+  })));
+
+  await Promise.all([
+    granulesModel.create(testGranule1),
+    granulesModel.create(testGranule2),
+  ]);
+
+  t.teardown(async () => {
+    granulesModel.delete({ granuleId: testGranule1.granuleId });
+    granulesModel.delete({ granuleId: testGranule2.granuleId });
+    await s3Utils.recursivelyDeleteS3Bucket(alternateBucket);
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 2,
+      failed: 0,
+      skipped: 0,
+      success: 2,
+    },
+    granulesSummary: {
+      dynamoRecords: 2,
+      failed: 0,
+      skipped: 0,
+      success: 2,
+    },
+  });
+  const records = await knex(tableNames.granules);
+  const fileRecords = await knex(tableNames.files);
+  t.is(records.length, 2);
+  t.is(fileRecords.length, 2);
+});
+
+test.serial('migrateGranulesAndFiles processes all non-failing granule records and does not process files of failling granule records', async (t) => {
```

Comment:
I've seen the word "failing" on my tests enough to know you have an extra "l" here 😄

Reply:
Ah thanks for catching that!